src/sounds.cpp

```diff
@@ -388,16 +393,790 @@ std::pair<std::vector<tripoint>, std::vector<tripoint>> sounds::get_monster_soun
+                       || monster_id == "mon_tankbot" ) {
+                play_variant_sound( "fire_gun", "m4a1", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_turret_bmg" ) {
+                play_variant_sound( "fire_gun", "m2browning", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_laserturret" ) {
+                play_variant_sound( "fire_gun", "laser_rifle", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            }
+        } else {
+            if( monster_id == "mon_turret" || monster_id == "mon_secubot" ) {
+                play_variant_sound( "fire_gun_distant", "smg", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_turret_rifle" || monster_id == "mon_chickenbot"
+                       || monster_id == "mon_tankbot" ) {
+                play_variant_sound( "fire_gun_distant", "rifle", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_turret_bmg" ) {
+                play_variant_sound( "fire_gun_distant", "rifle", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            } else if( monster_id == "mon_laserturret" ) {
+                play_variant_sound( "fire_gun_distant", "laser", heard_volume, angle, 0.8, 1.2 );
+                start_sfx_timestamp = std::chrono::high_resolution_clock::now();
+                return;
+            }
+        }
+    }
+}
+
+void sfx::generate_melee_soundfx( tripoint source, tripoint target, bool hit, bool targ_mon,
+                                  std::string material ) {
+    sound_thread * out = new sound_thread();
+    out->source = source;
+    out->target = target;
+    out->hit = hit;
+    out->targ_mon = targ_mon;
+    out->material = material;
+    //pthread_t thread1;
+    //pthread_create( &thread1, NULL, generate_melee_soundfx_thread, out );
+    return;
+}
+
+void *sfx::generate_melee_soundfx_thread( void * out ) {
+    std::srand( time( NULL ) );
+    std::this_thread::sleep_for( std::chrono::milliseconds( rng( 1, 2 ) ) );
+    sound_thread *in = ( sound_thread* ) out;
+    bool hit = in->hit;
+    bool targ_mon = in->targ_mon;
+    tripoint source = in->source;
+    tripoint target = in->target;
+    std::string material = in->material;
+    std::string variant_used;
+    int npc_index = g->npc_at( source );
+    if( npc_index == -1 ) {
```

Comment:
So .. uhm ... what's the difference between the two blocks here? Shouldn't the sound be the same for both player and any NPC? Why not use only one block:
```C++
player *p;
if( npc_index == -1 ) {
    p = &g->u;
} else {
    p = g->active_npc[npc_index];
}
// continue and use p
```